packages/@aws-cdk/aws-cloudwatch/lib/private/rendering.ts

```diff
@@ -173,6 +176,31 @@ export class MetricSet<A> {
     if (!entry.id && id) {
       entry.id = id;
       this.metricById.set(id, entry);
+      // if we don't have an id and this is a math expression
```

Comment:
@madeline-k I've implemented the auto assign id logic here. Rather than try and figure out if an id has already been manually assigned, I decided to instead just reserve an id format for auto assigned ids. 

Also, since this logic is at the metric level, it also applies to Alarms in addition to Graphs. Alarms already had their own similar auto assign id logic which this will overwrite.

Reply:
hmm I may have to rethink that since we have integration tests in other modules reference the Alarm id.

Reply:
@madeline-k I updated the logic so that it only applies to Graphs, but did end up having to update the kinesis integration tests.